projects/igniteui-angular/src/lib/date-range/igx-date-range.component.ts

```diff
@@ -0,0 +1,661 @@
+    private applyFocusOnClose() {
+        this.toggleDirective.onClosed
+            .pipe(takeUntil(this.$destroy))
+            .subscribe(() => (
+                this.single || this.projectedInputs.find(i => i instanceof IgxDateStartComponent) as IgxDateStartComponent
+            )?.setFocus());
+    }
+
+    private subscribeToDateEditorEvents() {
+        if (this.hasProjectedInputs) {
+            const start = this.projectedInputs.find(i => i instanceof IgxDateStartComponent) as IgxDateStartComponent;
+            const end = this.projectedInputs.find(i => i instanceof IgxDateEndComponent) as IgxDateEndComponent;
+            if (start && end) {
+                start.dateTimeEditor.valueChange
+                    .pipe(takeUntil(this.$destroy))
+                    .subscribe((date: Date) => {
+                        this.value = { start: date as Date, end: this.value?.end };
+                    });
+                end.dateTimeEditor.valueChange
+                    .pipe(takeUntil(this.$destroy))
+                    .subscribe((date: Date) => {
+                        this.value = { start: this.value?.start, end: date as Date };
+                    });
+                start.dateTimeEditor.validationFailed
+                    .pipe(takeUntil(this.$destroy))
+                    .subscribe((event: IgxDateTimeEditorEventArgs) => {
+                        this.value = { start: event.newValue as Date, end: this.value?.end };
+                    });
+                end.dateTimeEditor.validationFailed
+                    .pipe(takeUntil(this.$destroy))
+                    .subscribe((event: IgxDateTimeEditorEventArgs) => {
+                        this.value = { start: this.value?.start, end: event.newValue as Date };
+                    });
+            }
+        }
+    }
+
+    private subscribeToToggleEvents(): void {
+        this.toggleDirective.onOpened.pipe(takeUntil(this.$destroy)).subscribe(() => {
+            requestAnimationFrame(() => {
+                this.calendar.daysView.focusActiveDate();
+                this.onOpened.emit({ owner: this });
+            });
+        });
+        this.toggleDirective.onClosed.pipe(takeUntil(this.$destroy)).subscribe(() => {
+            if (this.value && !this.value.end) {
+                this.value = { start: this.value.start, end: this.value.start };
+                this.onClosed.emit({ owner: this });
+            }
+        });
+    }
+
+    private configPositionStrategy(): void {
+        this._positionSettings = {
+            openAnimation: fadeIn,
+            closeAnimation: fadeOut,
+            target: this.element.nativeElement
+        };
+        this._dropDownOverlaySettings.positionStrategy = new AutoPositionStrategy(this._positionSettings);
+    }
+
+    private configOverlaySettings(): void {
+        if (this.overlaySettings !== null) {
+            this._dropDownOverlaySettings = Object.assign({}, this._dropDownOverlaySettings, this.overlaySettings);
```

Comment:
I think this whole method could be dropped.
Currently, you're overwriting the default settings, which is not desired, as if the passed  custom `overlaySettings` can be changed more than once, you'll lose the default ones.
Furthermore, `configOverlaySettings` is only called once (in `ngAfterViewInit`) which means that the `overlaySettings` cannot be changed dynamically.

I suggest you make getters for the setting you'll be using to call toggle:
```
get dropDownOverlaySettings: OverlaySettings {
    return Object.assign({}, this._dropDownOverlaySettings, this.overlaySettings);
}
```
and just calling `this.dropDownOverlaySettings` instead of `this._dropDownOverlaySettings` everywhere. Same goes for the `_dialogOverlaySettings`.
This way, you'll ensure that the default settings are immutable